src/transaction/atomic_replication_helper.cpp

```diff
@@ -649,25 +655,27 @@ namespace cublog
 
 		if (prm_get_bool_value (PRM_ID_ER_LOG_PTS_ATOMIC_REPL_DEBUG))
 		  {
-		    dump ("sequence::can_purge - after LOG_SYSOP_END - LOG_SYSOP_END_LOGICAL_RUN_POSTPONE");
+		    dump ("sequence::can_purge(8) after erase");
 		  }
 	      }
 	  }
 	// part of scenario (4)
+	// part of scenario (5)
 	else if (LOG_SYSOP_END == last_entry.m_rectype
 		 && LOG_SYSOP_END_LOGICAL_UNDO == last_entry.m_sysop_end_type)
 	  {
 	    const atomic_log_entry_vector_type::const_iterator last_but_one_entry_it = (last_entry_it - 1);
 	    const atomic_log_entry &last_but_one_entry = *last_but_one_entry_it;
 
-	    if (!LSA_ISNULL (&last_entry.m_sysop_end_last_parent_lsa) &&
-		(last_but_one_entry.m_lsa >= last_entry.m_sysop_end_last_parent_lsa))
+	    if (last_but_one_entry.m_rectype == LOG_SYSOP_ATOMIC_START) //&&
+	      // !LSA_ISNULL (&last_entry.m_sysop_end_last_parent_lsa) &&
+	      // (last_but_one_entry.m_lsa >= last_entry.m_sysop_end_last_parent_lsa))
```

Comment:
is this change related to #3955 (which is now closed) ?

Reply:
In a way, yes.
In absence of the patch in #3955 I decided to drop the `lastparent_lsa` criterium. Thus the two encountered situations where the `lastparent_lsa` either is `(-1|-1)` or a valid value will be satisfied by the remaining condition.